AWSAuthSDK/AWSAuthSDKTestAppUITests/AWSHostedUIUserPoolTests.swift

```diff
@@ -58,34 +55,72 @@ class AWSHostedUIUserPoolTests: XCTestCase {
         // Check if all user details are present
         inspectTokenDetails(application: app)
         inspectCredentialDetails(application: app)
+    }
+
+    /// Test successful re-authentication in user pool when session expires using hosted UI and get user attributes
+    ///
+    /// - Given: Expire an user session after signing in
+    /// - When:
+    ///    - I try to get user attributes
+    /// - Then:
+    ///    - I should re-authenticate and successfully get user attributes
+    ///
+    func testHostedUIGetAttributesWhenSessionExpired() {
```

Comment:
Not sure whether we have captured all the steps here:

1. signIn a user (captured)
2. Verify user is signedIn (captured)
3. Fetch user attributes successfully (captured)
4. Invalidate token
5. Fetch user attribute again
6. User state change to signedOut
7. signIn the same user
8. User attribute returned successfully

Reply:
The steps for this test case is:

1. Sign Out a user if any
2. Sign Up and verify a new user
3. HostedUI Sign In a user
4. Verify user is signed In
5. Invalidate refresh token
6. Try to Fetch user attribute but fail, and hostedUI pops up automatically require to re-authenticate
7. Once successfully re-authenticated, user attributes are fetched successfully